jsx/Tabs.js

```diff
@@ -141,6 +141,123 @@ Tabs.defaultProps = {
+ */
+
+class VerticalTabs extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const hash = window.location.hash;
+    let activeTab = "";
+
+    /**
+     * Determine the initial active tab in this order
+     * 1. Try to infer from the URL, otherwise
+     * 2. Try to infer from the defaultTab prop, otherwise
+     * 3. Set to be the first tab of the list
+     */
+    if (this.props.updateURL && hash) {
+      activeTab = hash.substr(1);
+    } else if (this.props.defaultTab) {
+      activeTab = this.props.defaultTab;
+    } else if (this.props.tabs.length > 0) {
+      activeTab = this.props.tabs[0].id;
+    }
+
+    this.state = {
+      activeTab: activeTab
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+    this.getTabs = this.getTabs.bind(this);
+    this.getTabPanes = this.getTabPanes.bind(this);
+  }
+
+  handleClick(tabId, e) {
+    this.setState({activeTab: tabId});
+    this.props.onTabChange(tabId);
+
+    // Add tab href to URL querystring and scroll the page to top
+    if (this.props.updateURL) {
+      const scrollDistance = $('body').scrollTop() || $('html').scrollTop();
+      window.location.hash = e.target.hash;
+      $('html,body').scrollTop(scrollDistance);
+    }
+  }
+
+  getTabs() {
+    let tabs = (this.props.tabs).map(function(tab) {
+      let tabClass = this.state.activeTab === tab.id ? 'active' : null;
+      let href = "#" + tab.id;
+      let tabID = "tab-" + tab.id;
+      return (
+        <li
+          role="presentation"
+          className={tabClass}
+          key={tab.id}
+        >
+          <a id={tabID}
+             href={href}
+             role="tab"
+             data-toggle="tab"
+             onClick={this.handleClick.bind(null, tab.id)}
+          >
+            {tab.label}
+          </a>
+        </li>
+      );
+    }.bind(this));
+
+    return tabs;
+  }
+
+  getTabPanes() {
+    let tabPanes = React.Children.map(this.props.children, function(child, key) {
+      if (child) {
+        return React.cloneElement(child, {
+          activeTab: this.state.activeTab,
+          key: key
+        });
+      }
+    }.bind(this));
+
+    return tabPanes;
+  }
+
+  render() {
+    let tabs = this.getTabs();
+    let tabPanes = this.getTabPanes();
+    let tabStyle = {
+      marginLeft: 0,
+      marginBottom: '5px'
+    };
+
+    return (
+      <div>
+        <div className="tabbable col-md-3 col-sm-3">
+          <ul className="nav nav-pills nav-stacked" role="tablist" style={tabStyle}>
+            {tabs}
+          </ul>
+        </div>
+        <div className="tab-content col-md-9 col-sm-9">
+          {tabPanes}
+        </div>
+      </div>
+    );
+  }
+}
+VerticalTabs.propTypes = {
+  tabs: React.PropTypes.array.isRequired,
+  defaultTab: React.PropTypes.string,
+  updateURL: React.PropTypes.bool
+};
+VerticalTabs.defaultProps = {
+  onTabChange: function() {},
+  updateURL: false
```

Comment:
why does this default to false? I'm not even sure it's worth having an option rather than just always updating the URL..

Reply:
copy of line 141 - regular tabs